src/helm/benchmark/model_deployment_registry.py

```diff
@@ -27,68 +35,90 @@ class ModelDeployment:
     window_service_spec: Optional[WindowServiceSpec] = None
-    """Specification for instantiating the window service for this model deployment"""
 
+    # Maximum sequence length for this model deployment.
     max_sequence_length: Optional[int] = None
-    """Maximum sequence length for this model deployment."""
 
+    # Maximum request length for this model deployment.
+    # If unset, defaults to the same value as max_sequence_length.
     max_request_length: Optional[int] = None
-    """Maximum request length for this model deployment.
-
-    If unset, defaults to the same value as max_sequence_length."""
 
+    # The max length of the model input and output tokens.
+    # Some models (like Anthropic/Claude and Megatron) have a specific limit sequence length + max_token.
+    # If unset, defaults to INT_MAX (i.e. bo limit).
     max_sequence_and_generated_tokens_length: Optional[int] = None
-    """The max length of the model input and output tokens.
 
-    Some models (like Anthropic/Claude and Megatron) have a specific limit sequence length + max_token.
+    # Whether this model deployment is deprecated.
+    deprecated: bool = False
 
-    If unset, defaults to INT_MAX (i.e. no limit)."""
+    @property
+    def host_organization(self) -> str:
+        """
+        Extracts the host group from the model deployment name.
+        Example: "huggingface" from "huggingface/t5-11b"
+        This can be different from the creator organization (for example "together")
+        """
+        return self.name.split("/")[0]
+
+    @property
+    def engine(self) -> str:
+        """
+        Extracts the model engine from the model deployment name.
+        Example: 'ai21/j1-jumbo' => 'j1-jumbo'
+        """
+        return self.name.split("/")[1]
 
 
 @dataclass(frozen=True)
 class ModelDeployments:
     model_deployments: List[ModelDeployment]
 
 
-_name_to_model_deployment: Dict[str, ModelDeployment] = {}
+ALL_MODEL_DEPLOYMENTS: List[ModelDeployment] = []
+DEPLOYMENT_NAME_TO_MODEL_DEPLOYMENT: Dict[str, ModelDeployment] = {
+    deployment.name: deployment for deployment in ALL_MODEL_DEPLOYMENTS
+}
 
 
+# ===================== REGISTRATION FUNCTIONS ==================== #
 def register_model_deployment(model_deployment: ModelDeployment) -> None:
-    hlog(f"Registered model deployment {model_deployment.name}")
-    _name_to_model_deployment[model_deployment.name] = model_deployment
-
-    # Auto-register a model with this name if none exists
-    model_name = model_deployment.model_name or model_deployment.name
-    if model_name not in MODEL_NAME_TO_MODEL:
-        model = Model(
-            group="unknown",
-            name=model_name,
-            tags=[TEXT_MODEL_TAG, FULL_FUNCTIONALITY_TEXT_MODEL_TAG],
-        )
-        MODEL_NAME_TO_MODEL[model_name] = model
-        ALL_MODELS.append(model)
-        hlog(f"Registered default metadata for model {model_name}")
```

Comment:
I think you have to move this logic into `huggingface_registration.py`, otherwise the `--enable-huggingface-models` functionality will break.

Reply:
Added but no way to test this